log_downloader_service.py

```diff
@@ -0,0 +1,87 @@
+import os
+import time
+import shutil
+from pathlib import Path
+from loguru import logger
+from git import Repo
+import wmi
+
+
+REPO_PATH = Path(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))) # TODO: change
+LOGS_DIR = REPO_PATH / "logs"
+LOGS_DIR.mkdir(exist_ok=True)
+
+
+def commit_log(repo, log_file):
+    try:
+        repo.index.add([str(log_file)])
+        repo.index.commit(f"Add log file: {log_file.name}")
+        logger.info(f"Committed {log_file.name} to repository")
+    except Exception as e:
+        logger.info(f"Error committing {log_file.name}: {e}")
+
+
+def get_file_signature(file_path):
+    return file_path.name, file_path.stat().st_size
+
+
+def is_file_downloaded(log_file):
+    file_name = log_file.name
+    file_size = log_file.stat().st_size
+
+    for existing_file in LOGS_DIR.glob("*.wpilog"):
+        if (existing_file.name == file_name and
+                existing_file.stat().st_size == file_size):
+            return True
+    return False
+
+def get_usb_drives():
+    drives = set()
+    c = wmi.WMI()
+
+    for drive in c.Win32_LogicalDisk():
+        if drive.DriveType == 2: # removable
+            drives.add(Path(f"{drive.DeviceID}/"))
+
+    return drives
+
+def download_logs(drive_path, repo):
+    try:
+        for log_file in drive_path.glob("**/*.wpilog"):
+            if not is_file_downloaded(log_file):
+                dest_path = LOGS_DIR / log_file.name
+                logger.info(f"Copying {log_file.name} ({log_file.stat().st_size} bytes)")
+                shutil.copy2(log_file, dest_path)
+                commit_log(repo, dest_path)
+            else:
+                logger.info(f"Skipping {log_file.name} - already exists")
+    except Exception as e:
+        logger.info(f"Error accessing {drive_path}: {e}")
```

Comment:
1. It's better to wrap only the parts of code we suspect could raise an exception with `try`-`except`. If there's e.g. a problem with committing a file, we would still see the genetic "error accessing" log which isn't accurate and won't help us understand the problem.
2. It's better to have the function download one log file and call it from another for loop.

Reply:
bdbe0327f2cf35cef72aafd47fb4e38089e8b0c7

Reply:
You still wrap many lines in one try block. It should be this way:
```py
    for log_file in drive_path.glob("**/*.wpilog"):
        try:
            shutil.copy2(log_file, dest_path)
        except OSError as error:
            logger.error(f"Failed to copy {log_file.name}: {error}")
            continue
        logger.info(f"Copied {log_file.name} ({log_file.stat().st_size} bytes)")
        commit_log(repo, dest_path)
```
1. `.wpilog` should be constant.
2. `copy2()` is the only line that can fail, so it's the only one that should be in the try block. Errors that occur during log committing should be handled in that function.
3. Don't shorten: Use `error` and not `error`.
4. Log finished tasks ("copied", not "copying"), and only after the task finished successfully (so not before copying the file).
5. Note that you do not address #6.

Reply:
2. 72e81c7edbbf1cbde2cb76ffef5692c44fec44ef

Reply:
What did you mean in 3? Also 6 was fixed some time ago.

Reply:
I don't think type hints are necessary.

Reply:
1. OK
2. Not fixed... check what you committed.
3. Whoops. I meant use `error`, not `e` for the variable name.
4. About 6 OK. I looked at a old version.
5. Type hints are pretty standard today (e.g. we use them everywhere in the army). They also help the IDE provide code completion.

Reply:
5. I think it's best to just download all log files, It's not that many logs anyway since this will run constantly. So i think downloading all of them no matter what is okay.

Reply:
I would at least save them in a different directory, or add some prefix to the file name, so when sorted by name, the matches log files are at the top.
How do you prevent opening the visualizer for non-match files?

Reply:
f833d13cdd737522bd39684aaab95b8eedc05cf8